### PR TITLE
Boost dependency

### PR DESCRIPTION
**Pull request by jgphpc**
Fixes https://github.com/MeteoSwiss-APN/comm_overlap_bench/issues/11

### CODE REVIEW

**Review by pspoerri — CHANGES_REQUESTED** [review]
Just replace the `boost::shared_ptr` with `std::shared_ptr`. It makes no sense to do something like:
```
#ifndef _NO_BOOST
#include <boost/make_shared.hpp>
#else
#include <memory>
#endif
```

**Comment by jgphpc** [comment]
Well, i still want to be able to call Boost timers if needed. Do you mean that `#include <memory>` will work in both cases ?

**Comment by pspoerri** [comment]
The boost timers are fine: They are enabled and guarded  with the `ENABLE_TIMER` flag. Unless, I did some mistake when I rewrote it. The `boost::shared_ptr` was still in there because I simply forgot to replace it. Also `boost::shared_ptr` and `std::shared_ptr` essentially do the same thing and will not have an impact on the timers.

If we remove the `boost::shared_ptr` we can't target older compilers (< GCC 4.9) anymore, since `std::shared_ptr` is supported since C++11. Looking at our current operational setup, we use some features of C++11 anyway, hence this is not a big issue. Also I'm already using `std::shared_ptr` in some other location.

**Comment by pspoerri** [comment]
And yes, to answer your question. `#include <memory>` will work in both cases. Make sure, that you remove the `boost::shared_ptr` occurences from places like here:
```
#ifndef _NO_BOOST
     std::vector< boost::shared_ptr< SwapDataField<IJKRealField> > >& horiDiffFields() { return horiDiffFields_; }
#else
    std::vector< std::shared_ptr< SwapDataField<IJKRealField> > >& horiDiffFields() { return horiDiffFields_; }
#endif
```

Hence I want to see something like:
```
#include <memory>

// [...] snip
    std::vector< std::shared_ptr< SwapDataField<IJKRealField> > >& horiDiffFields() { return horiDiffFields_; }

// [...] snip
```
instead.

File: src/main.cpp [hunk 1]
```diff
@@ -1,7 +1,9 @@
 #include <stdexcept>
 #include <iostream>
 #include <cmath>
+#ifndef _NO_BOOST
 #include <boost/lexical_cast.hpp>
+#endif
```
Comment (pspoerri) [inline src/main.cpp]: I don't think the `lexical_cast` is needed. You can remove it as well.

File: src/HorizontalDiffusionSA.cpp [hunk 2]
```diff
@@ -1,4 +1,6 @@
+#ifndef _NO_BOOST
 #include <boost/preprocessor/repetition/repeat.hpp>
+#endif
```
Comment (pspoerri) [inline src/HorizontalDiffusionSA.cpp]: The repeat is not needed here. You can remove it as well.

**Comment by pspoerri** [comment]
I feel terribly sorry: I should have rewritten the code from scratch when I removed the STELLA dependencies. But I did not have the time for it. Right now the code is a big mess and I want to avoid expanding the mess by adding additional flags that are not needed.

**Comment by jgphpc** [comment]
This should be good now.

**Review by pspoerri — APPROVED** [review]

**Pull request merged**